File: src/main/java/ch/admin/seco/jobroom/dto/RegisteredJobPosition.java
package ch.admin.seco.jobroom.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.HashSet;
import java.util.Set;

@JsonIgnoreProperties(ignoreUnknown = true)
public class RegisteredJobPosition extends JobPosition {

    private String egovId;

    private final Set<SelfLink> links = new HashSet<SelfLink>();

    private String state;

    private EmploymentOffice office;

    public String getEgovId() {
        return egovId;
    }

    public void setEgovId(String egovId) {
        this.egovId = egovId;
    }

    public Set<SelfLink> getLinks() {
        return links;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public EmploymentOffice getOffice() {
        return office;
    }

    public void setOffice(EmploymentOffice office) {
        this.office = office;
    }
}

File: src/main/java/ch/admin/seco/jobroom/dto/RegisteredJobPosition.kt
package ch.admin.seco.jobroom.dto

// FIXME consider composition/mixin style instead of inheritance ?
class RegisteredJobPosition(

        val id: Long,

        // TODO: not necessary to keep this reference!
        private val job: JobPosition

    ) : JobPosition(

        title = job.title,
        description = job.description,
        countryCode = job.countryCode,
        city = job.city,
        zip = job.zip,
        startImmediate = job.startImmediate,
        languageSkills = job.languageSkills

    )

//    val links: Set<SelfLink> = HashSet()
//
//    var state: String? = null

